--- ../src_base/minecraft/net/minecraft/client/renderer/entity/RenderManager.java
+++ ../src_work/minecraft/net/minecraft/client/renderer/entity/RenderManager.java
@@ -88,6 +88,8 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.util.ReportedException;
 import net.minecraft.world.World;
+import net.minecraftforge.client.event.EntityRenderEvent;
+import net.minecraftforge.common.MinecraftForge;
 import org.lwjgl.opengl.GL11;
 
 @SideOnly(Side.CLIENT)
@@ -221,12 +223,14 @@
 
         if (par4EntityLiving.isPlayerSleeping())
         {
-            int i = par1World.getBlockId(MathHelper.floor_double(par4EntityLiving.posX), MathHelper.floor_double(par4EntityLiving.posY), MathHelper.floor_double(par4EntityLiving.posZ));
-
-            if (i == Block.bed.blockID)
+            int x = MathHelper.floor_double(par4EntityLiving.posX);
+            int y = MathHelper.floor_double(par4EntityLiving.posY);
+            int z = MathHelper.floor_double(par4EntityLiving.posZ);
+            Block block = Block.blocksList[par1World.getBlockId(x, y, z)];
+
+            if (block != null && block.isBed(par1World, x, y, z, par4EntityLiving))
             {
-                int j = par1World.getBlockMetadata(MathHelper.floor_double(par4EntityLiving.posX), MathHelper.floor_double(par4EntityLiving.posY), MathHelper.floor_double(par4EntityLiving.posZ));
-                int k = j & 3;
+                int k = block.getBedDirection(par1World, x, y, z);;
                 this.playerViewY = (float)(k * 90 + 180);
                 this.playerViewX = 0.0F;
             }
@@ -288,6 +292,11 @@
         try
         {
             render = this.getEntityRenderObject(par1Entity);
+
+            if (!MinecraftForge.EVENT_BUS.post(new EntityRenderEvent(render, par1Entity, par2, par4, par6, par8, par9)))
+            {
+                return;
+            }
 
             if (render != null && this.renderEngine != null)
             {
